test/e2e/operator_test.go

```diff
@@ -793,6 +793,74 @@ func TestDNSOverTLSForwarding(t *testing.T) {
 	}
 }
 
+func TestDNSOverTLSToleratesMissingSourceCM(t *testing.T) {
+	cl, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Ensure that DNS is stable before starting the test.
+	if err := waitForDNSConditions(t, cl, 5*time.Minute, dnsName, defaultAvailableDNSConditions...); err != nil {
+		t.Errorf("expected default DNS pods to be available: %v", err)
+	}
+
+	// Update cluster DNS forwarding with the missing configmap.
+	defaultDNS := &operatorv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: operatorcontroller.DefaultDNSController}, defaultDNS); err != nil {
+		t.Fatalf("failed to get default dns: %v", err)
+	}
+
+	upstream := operatorv1.Server{
+		Name:  "test",
+		Zones: []string{"tls.com"},
+		ForwardPlugin: operatorv1.ForwardPlugin{
+			TransportConfig: operatorv1.DNSTransportConfig{
+				Transport: operatorv1.TLSTransport,
+				TLS: &operatorv1.DNSOverTLSConfig{
+					ServerName: "dns.tls.com",
+					CABundle:   configv1.ConfigMapNameReference{Name: "missing-cm"},
+				},
+			},
+			Upstreams: []string{"1.1.1.1"},
+		},
+	}
+
+	defaultDNS.Spec.Servers = []operatorv1.Server{upstream}
+	if err := cl.Update(context.TODO(), defaultDNS); err != nil {
+		t.Fatalf("failed to update dns %s: %v", defaultDNS.Name, err)
+	}
+	t.Cleanup(func() {
+		defaultDNS = &operatorv1.DNS{}
+		if err := cl.Get(context.TODO(), types.NamespacedName{Name: "default"}, defaultDNS); err != nil {
+			t.Fatalf("failed to get default dns: %v", err)
+		}
+		if len(defaultDNS.Spec.Servers) != 0 {
+			// dnses.operator/default has a nil spec by default.
+			defaultDNS.Spec = operatorv1.DNSSpec{}
+			if err := cl.Update(context.TODO(), defaultDNS); err != nil {
+				t.Fatalf("failed to update dns %s: %v", defaultDNS.Name, err)
+			}
+		}
+	})
+
+	// Verify that default DNS pods are all available before inspecting them.
+	if err := waitForDNSConditions(t, cl, 5*time.Minute, dnsName, defaultAvailableDNSConditions...); err != nil {
+		t.Errorf("expected default DNS pods to be available: %v", err)
+	}
+
+	dnsOperatorPods, err := getClusterDNSOperatorPods(cl)
+	if err != nil {
+		t.Errorf("unable to get operator pods: #{err}")
```

Comment:
This `#{err}` looks like a goland artifact.

Reply:
Fixed in https://github.com/openshift/cluster-dns-operator/pull/321/commits/9dd51de4c823be08b33d55b182652d9db3dfc6e8